src/main/java/org/example/service/DepartmentService.java

```diff
@@ -0,0 +1,57 @@
+package org.example.service;
+
+import org.example.DTO.DepartmentDTO;
+import org.example.model.DepartmentModel;
+import org.example.repository.DepartmentRepository;
+import org.example.Exception.ResourceNotFoundException;
+import org.example.Exception.BadRequestException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class DepartmentService {
+
+    @Autowired
+    private DepartmentRepository departmentRepository;
+
+    private DepartmentDTO convertToDTO(DepartmentModel departmentModel) {
+        return new DepartmentDTO(departmentModel.getId(), departmentModel.getName());
+    }
+
+    private DepartmentModel convertToEntity(DepartmentDTO departmentDTO) {
+        return new DepartmentModel(departmentDTO.getId(), departmentDTO.getName());
+    }
+
+    public DepartmentDTO saveDepartment(DepartmentDTO departmentDTO) {
+        if (departmentDTO == null || departmentDTO.getName().isEmpty()) {
```

Comment:
don't need that check once you use @Valid and you can use @Notnull on the dto in the controller also